src/cmap/connection_pool.ts

```diff
@@ -437,7 +525,31 @@ function createConnection(pool: ConnectionPool, callback?: Callback<Connection>)
       connection.on(event, (e: any) => pool.emit(event, e));
     }
 
-    pool.emit(ConnectionPool.CONNECTION_POOL_CREATED, new ConnectionCreatedEvent(pool, connection));
+    pool.emit(ConnectionPool.CONNECTION_CREATED, new ConnectionCreatedEvent(pool, connection));
+
+    if (pool.loadBalanced) {
+      const onPin = (pinType: string) => {
+        pool[kMetrics].markPinned(pinType);
+      };
+      const onUnpin = (pinType: string) => {
+        pool[kMetrics].markUnpinned(pinType);
+      };
+      connection.on('pinned', onPin);
+      connection.on('unpin', onUnpin);
```

Comment:
Can we make these constants on the Connection class like our other errors?

Reply:
Constants on `Connection` now.

Reply:
I see the constants but can we make use of them here?

Reply:
Thanks for catching that - they use it now.